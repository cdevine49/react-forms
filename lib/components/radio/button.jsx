import React from 'react';
import Label from '../label';
import setClassName from '../../helpers/setClassName';

const RadioButton = (
  {
    className,
    containerProps: {
      className: containerClass,
      ...containerProps
    },
    id,
    label,
    labelProps: {
      className: labelClass,
      ...labelProps
    },
    onChange: onChangeCallback,
    value,
    ...props
  }
) => {
  !id && console.warn("RadioButton expects an id");
  !value && console.warn("RadioButton expects a value");

  const onChange = e => {
    e.stopPropagation();
    onChangeCallback();
  };

  const onClick = e => {
    e.stopPropagation();
  };

  return (
    <div className={ setClassName(['radio-button', containerClass]) } onClick={ onChange } { ...containerProps }>
      <input
        id={ id }
        type='radio'
        className={ setClassName(['radio-input', className]) }
        onChange={ onChange }
        onClick={ onClick }
        value={ value }
        { ...props }
        />
      <Label htmlFor={ id } className={ setClassName(['radio-label', labelClass]) } { ...labelProps } >{ label || value }</Label>
    </div>
  );
}
RadioButton.defaultProps = {
  containerProps: {},
  id: '',
  labelProps: {},
  value: ''
}

export default RadioButton;
