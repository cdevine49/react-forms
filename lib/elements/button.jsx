import styled from 'styled-components';

export const Button = styled.button``;

export const Submit = Button.extend.attrs({
  type: 'submit'
})`
  align-self: flex-start;
  padding: 6px 12px;
  line-height: 20px;
  vertical-align: middle;
  text-align: center;
  cursor: pointer;
  color: #fff;
  text-shadow: 0 -1px 0 rgba(0,0,0,0.15);
  background-color: #6cc644;
  background-image: linear-gradient(#91dd70, #55ae2e);
  border: 1px solid #5aad35;
  border-radius: 3px;
`;

Submit.defaultProps = {
  children: 'Submit'
}