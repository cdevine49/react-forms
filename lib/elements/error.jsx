import styled from 'styled-components';

const Error = styled.p`
  margin-top: 6.5px;
  color: #dd4b39;
`;

export default Error;