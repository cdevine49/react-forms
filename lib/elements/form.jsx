import styled from 'styled-components';

const Form = styled.form`
  display: flex;
  flex-direction: column;
  width: 400px;
  padding: 20px;
`;

export default Form;

Form.defaultProps = {
  onSubmit: () => {}
}