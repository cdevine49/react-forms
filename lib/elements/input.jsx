import styled from 'styled-components';

const Input = styled.input`
  background: #FFFFFF;
  line-height: 20px;
  vertical-align: middle;
  padding: 6px 8px;
  color: #222;
  vertical-align: middle;
  border: 1px solid #ddd;
  border-radius: 3px 3px;
  box-shadow: inset 0 1px 2px rgba(0,0,0,0.075);
  background-repeat: no-repeat;
  background-position: right 8px center;
  outline: none;

  &:hover {
    border-color: #b9b9b9;
    border-top-color: #a0a0a0;
  }

  &:focus {
    border-color: #4d90fe;
  }
  border-color: ${props => (props.error && '#dd4b39') || (props.focus && '#dd4b39')};
`;

export default Input;