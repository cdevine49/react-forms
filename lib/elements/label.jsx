import styled from 'styled-components';

export const Label = styled.label`
  color: #222;
  font-weight: bold;
  margin-bottom: 3px;
`;

export const Required = styled.span`

`;
