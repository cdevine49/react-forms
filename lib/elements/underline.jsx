import styled from 'styled-components';

const Underline = styled.p`
  min-height: 17px;
  margin: 4px 0 2px;
  font-size: 12px;
  color: #767676;
`;

export default Underline;