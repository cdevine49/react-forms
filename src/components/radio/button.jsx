import React from 'react';
import Label from '../label';
import setClassName from '../../helpers/setClassName';

const RadioButton = (
  {
    className,
    containerProps: {
      containerClass,
      ...containerProps
    },
    id,
    label,
    labelProps: {
      className: labelClass,
      ...labelProps
    },
    onChange,
    value,
    ...props
  }
) => {
  !id && console.warn("RadioButton expects an id");
  !value && console.warn("RadioButton expects a value");

  return (
    <div className={ setClassName(['radio-button', containerClass]) } onClick={ onChange } { ...containerProps }>
      <input
        id={ id }
        type='radio'
        className={ setClassName(['radio-input', className]) }
        onChange={ onChange }
        value={ value }
        { ...props }
        />
      <Label htmlFor={ id } className={ setClassName(['radio-label', labelClass]) } { ...labelProps } >{ label || value }</Label>
    </div>
  );
}
RadioButton.defaultProps = {
  containerProps: {},
  id: '',
  labelProps: {},
  value: null
}

export default RadioButton;
