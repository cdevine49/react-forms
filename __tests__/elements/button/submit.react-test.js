import React from 'react';
import { Submit } from '../../../lib/elements/button';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

test('renders with default props', () => {
  const tree = renderer.create(
    <Submit
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

test('accepts new props', () => {
  const onFocus = jest.fn();
  const tree = renderer.create(
    <Submit
      id="submitId"
      className="classy submit"
      onFocus={ onFocus }
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

test('default children can be overridden', () => {
  const tree = renderer.create(
    <Submit>New Child</Submit>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
