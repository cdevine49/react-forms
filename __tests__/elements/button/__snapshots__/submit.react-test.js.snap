// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`accepts new props 1`] = `
.c0 {
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  padding: 6px 12px;
  line-height: 20px;
  vertical-align: middle;
  text-align: center;
  cursor: pointer;
  color: #fff;
  text-shadow: 0 -1px 0 rgba(0,0,0,0.15);
  background-color: #6cc644;
  background-image: linear-gradient(#91dd70,#55ae2e);
  border: 1px solid #5aad35;
  border-radius: 3px;
}

<button
  className="classy submit c0"
  id="submitId"
  onFocus={[Function]}
  type="submit"
>
  Submit
</button>
`;

exports[`default children can be overridden 1`] = `
.c0 {
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  padding: 6px 12px;
  line-height: 20px;
  vertical-align: middle;
  text-align: center;
  cursor: pointer;
  color: #fff;
  text-shadow: 0 -1px 0 rgba(0,0,0,0.15);
  background-color: #6cc644;
  background-image: linear-gradient(#91dd70,#55ae2e);
  border: 1px solid #5aad35;
  border-radius: 3px;
}

<button
  className="c0"
  type="submit"
>
  New Child
</button>
`;

exports[`renders with default props 1`] = `
.c0 {
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  padding: 6px 12px;
  line-height: 20px;
  vertical-align: middle;
  text-align: center;
  cursor: pointer;
  color: #fff;
  text-shadow: 0 -1px 0 rgba(0,0,0,0.15);
  background-color: #6cc644;
  background-image: linear-gradient(#91dd70,#55ae2e);
  border: 1px solid #5aad35;
  border-radius: 3px;
}

<button
  className="c0"
  type="submit"
>
  Submit
</button>
`;
